test/Rx/Functional/Observable/ForkJoinObservableTest.php

```diff
@@ -0,0 +1,468 @@
+     */
+    public function forkjoin_empty_throw()
+    {
+        $error = new \Exception();
+
+        $e0 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(230),
+        ]);
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(210, $error),
+            onCompleted(250)
+        ]);
+
+        $xs = new ForkJoinObservable([$e0, $e1], function($a, $b) {
+            return $a + $b;
+        });
+
+        $results = $this->scheduler->startWithCreate(function () use ($xs) {
+            return $xs;
+        });
+
+        $this->assertMessages([
+            onError(210, $error),
+        ], $results->getMessages());
+    }
+
+    /**
+     * @test
+     */
+    public function forkjoin_throw_empty()
+    {
+        $error = new \Exception();
+
+        $e0 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(210, $error),
+            onCompleted(230),
+        ]);
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(250)
+        ]);
+
+        $xs = new ForkJoinObservable([$e0, $e1], function($a, $b) {
+            return $a + $b;
+        });
+
+        $results = $this->scheduler->startWithCreate(function () use ($xs) {
+            return $xs;
+        });
+
+        $this->assertMessages([
+            onError(210, $error),
+        ], $results->getMessages());
+    }
+
+    /**
+     * @test
+     */
+    public function forkjoin_return_throw()
+    {
+        $error = new \Exception();
+
+        $e0 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(210, 2),
+            onCompleted(230),
+        ]);
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(220, $error),
+            onCompleted(250)
+        ]);
+
+        $xs = new ForkJoinObservable([$e0, $e1], function($a, $b) {
+            return $a + $b;
+        });
+
+        $results = $this->scheduler->startWithCreate(function () use ($xs) {
+            return $xs;
+        });
+
+        $this->assertMessages([
+            onError(220, $error),
+        ], $results->getMessages());
+    }
+
+    /**
+     * @test
+     */
+    public function forkjoin_throw_return()
+    {
+        $error = new \Exception();
+
+        $e0 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(220, $error),
+            onCompleted(230),
+        ]);
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(210, 2),
+            onCompleted(250)
+        ]);
+
+        $xs = new ForkJoinObservable([$e0, $e1], function($a, $b) {
+            return $a + $b;
+        });
+
+        $results = $this->scheduler->startWithCreate(function () use ($xs) {
+            return $xs;
+        });
+
+        $this->assertMessages([
+            onError(220, $error),
+        ], $results->getMessages());
+    }
+}
```

Comment:
Can you add some dispose tests?  If you dispose before they all complete, all of the subscription should stop.  ie: https://github.com/ReactiveX/RxPHP/blob/master/test/Rx/Functional/Operator/DelayTest.php#L318